### DISCUSSION TITLE
Correct use of ORDER BY?

### DISCUSSION THREAD

**Discussion by ships** [oxigraph/oxigraph#590] [Q&A] [answered]
Running latest oxigraph docker image, I submit a SPARQL query like that:

```
PREFIX pwo: <http://purl.org/spar/pwo/>
PREFIX pso: <http://purl.org/spar/pso/>
PREFIX dm: <https://docmaps-project.github.io/ex/>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
SELECT * WHERE {
  ?map a pwo:Workflow .
  ?map pwo:hasStep ?step .
  ?step ?any ?obj .
} LIMIT 10 ORDER BY ?obj
```

but I get an error like this: `error at 10:18: expected one of OFFSET, VALUES`.

Adding `OFFSET 0` at the end does not make a difference.

Am I misunderstanding the correct use of ORDER BY? I am new to sparql -- oxigraph is the only sparql processor I have used with real data.

**Comment by Tpt** [comment, marked as answer]
Hi! Thank you for using Oxigraph! The error is because `LIMIT` must be after `ORDER BY` in SPARQL. Sorry for the very bad error message, improving them is in my TODO.

**Reply by ships** [reply]
Aha, ok that solves. Thanks!
